### PR TITLE
RPC-123-deploy-CryptoPunc

### PR DESCRIPTION
**Pull request by SKonstantinS**
add migration for CryptoPunc, proxyCryptoPunc

### CODE REVIEW

File: deploy/migrations/14_initial_CryptoPunks.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const settings = getSettings(network);
+  let cryptoPunksMarket;
+
+  if (settings.deploy_CryptoPunks) {
+    await deployer.deploy(CryptoPunksMarket, { gas: 4500000 });
+    cryptoPunksMarket = await CryptoPunksMarket.deployed();
+  } else {
+    cryptoPunksMarket = await CryptoPunksMarket.at(settings.address_CryptoPunks);
+  }
+  console.log("deployed cryptoPunksMarket:",  cryptoPunksMarket.address);
```
Comment (disenotov) [inline deploy/migrations/14_initial_CryptoPunks.js:18]: i think "deployed" is not needed here

File: deploy/migrations/config.js [hunk 2]
```diff
@@ -19,15 +21,23 @@ const mainnet = {
     artblocksAddress: "0x6C093Fe8bc59e1e0cAe2Ec10F0B717D3D182056B"
   },
   communityWallet: "0x1cf0df2a5a20cd61d68d4489eebbf85b8d39e18a",
+  deploy_CryptoPunks: false,
+  address_CryptoPunks: "0xb47e3cd837dDF8e4c57F05d70Ab865de6e193BBB",
 }
 const ropsten = {
   communityWallet: "0xe627243104a101ca59a2c629adbcd63a782e837f",
+  deploy_CryptoPunks: true,
+  address_ownerTestCryptoPunks: "0xc66d094ed928f7840a6b0d373c1cd825c97e3c7c",
```
Comment (disenotov) [inline deploy/migrations/config.js:30]: what is this address? 0xc66d094ed928f7840a6b0d373c1cd825c97e3c7c

File: deploy/migrations/14_initial_CryptoPunks.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  await punkTransferProxy.addOperator(exchangeV2.address);
+
+  await exchangeV2.setTransferProxy(CRYPTO_PUNKS, punkTransferProxy.address);
+  await setTestCryptoPunks(settings, punkTransferProxy.address);
+};
+
+async function setTestCryptoPunks(_settings, _punkTransferProxy) {
+  if (_settings.deploy_CryptoPunks) {
+    const cryptoPunksMarket = await CryptoPunksMarket.deployed();
```
Comment (disenotov) [inline deploy/migrations/14_initial_CryptoPunks.js:35]: this could be sent as an argument to this function

**Review by disenotov — CHANGES_REQUESTED** [review]
few minor changes + address_ownerTestCryptoPunks should be correct for every test network

File: deploy/migrations/14_initial_CryptoPunks.js (hunk 1 above)
Comment (SKonstantinS) [reply deploy/migrations/14_initial_CryptoPunks.js:18]: ok

File: deploy/migrations/config.js (hunk 2 above)
Comment (SKonstantinS) [reply deploy/migrations/config.js:30]: ok, add

File: deploy/migrations/14_initial_CryptoPunks.js (hunk 3 above)
Comment (SKonstantinS) [reply deploy/migrations/14_initial_CryptoPunks.js:35]: ok done

**Pull request merged**